thorlcr/master/thgraphmanager.cpp

```diff
@@ -1170,41 +1205,45 @@ void thorMain(ILogMsgHandler *logHandler, const char *wuid, const char *graphNam
 
                 if (lingerPeriod)
                 {
+                    Owned<IRemoteConnection> multiJobLingerInstanceConn;
+                    VStringBuffer multiJobLingerXPath("/Status/ThorLinger/%s", globals->queryProp("@name"));
+                    if (multiJobLinger)
+                    {
+                        Owned<IRemoteConnection> conn = querySDS().connect(multiJobLingerXPath, myProcessSession(), RTM_CREATE_QUERY | RTM_LOCK_WRITE, 5*60*1000);
+
+                        StringBuffer instance;
+                        queryMyNode()->endpoint().getUrlStr(instance);
+                        VStringBuffer entryXPath("%s/Thor%" I64F "u", multiJobLingerXPath.str(), myProcessSession());
+                        multiJobLingerInstanceConn.setown(querySDS().connect(entryXPath, myProcessSession(), RTM_CREATE_QUERY | RTM_DELETE_ON_DISCONNECT | RTM_LOCK_WRITE, 5*1000));
```

Comment:
Some comments on the lifetime would help when reading/returning to this code.

Reply:
will add